go-controller/pkg/util/batching/batch.go

```diff
@@ -21,3 +21,37 @@ func Batch[T any](batchSize int, data []T, eachFn func([]T) error) error {
 	}
 	return nil
 }
+
+func BatchMap[T any](batchSize int, data map[string][]T, eachFn func(map[string][]T) error) error {
+	if batchSize < 1 {
+		return fmt.Errorf("batchSize should be > 0, got %d", batchSize)
+	}
+	batchCounter := 0
+	batchMap := map[string][]T{}
+
+	for key, value := range data {
+		// adding nex value will exceed batch size, exec eachFn and start the next batch
+		if batchCounter+len(value) > batchSize {
```

Comment:
Why don't you iterate value as well? Then you wouldn't have the error below. If you find it important that receivers wont receive the same key with partial values on two or more consecutive iterations (which would be strange because it would not be a problem inherent to this function itself) then perhaps describe the behavior in a function doc.

Reply:
I default to the simpler implementation + a comment to not over-engineer it, but lmk if you think it's not the best thing to do now

Reply:
If you want my opinion, my suggestion is how I would expect `BatchMap` to behave, independent of usage. But what I wanted to say is that if we have different opinions on how it should behave, I am ok with picking either of them but we should document it like any other funciton is typically documented:

```
// BatchMap <explain what it does>
func BatchMap[T any](batchSize int, data map[string][]T, eachFn func(map[string][]T) error) error {
```

I don't think the current doc explains what it does in enough detail, it just documents one constraint of the function.

Reply:
I tried to add a little doc comment, hopefully it is better now

Reply:
What about
```
// BatchMap splits values in `data` in batches of size `batchSize` or lower, and calls `eachFn` for each batch.
// Values of a specific key are not split in different batches. If a specific key has more values than `batchSize`, 
// an error is returned
```

But now that I think of it in detail, what is a user supposed to do if it has more values than `batchSize` for a key. In our specific usage, you are guesstimating that you won't have more than 80k ACLs for a namespace? I have the feeling that we are limiting the usability of the function for a use that we don't have (e.g. the one you mentioned `if I was about to delete some namespaced port group after ACLs are migrated somewhere else`) and risking a potential use case that we might have (more that 80k ACLs for a namespace because I am not aware that is hard limited anywhere); if that makes sense.

Reply:
I like your BatchMap description, thank you!
I think I lost a comment somewhere between changes, because every namespace may only have 8K ACLs for EF. So I am guessing indeed, what is more likely to happen with the future cases: someone has a huge map value or someone needs to ensure map values are not split? 
Maybe at this point first option makes more sense, also because it may result in less transactions potentially. So I will just change it (but I need to rewrite that nice comment you shared :'( )